src/ui/SwiftOutputChannel.ts

```diff
@@ -15,11 +15,56 @@
 import * as vscode from "vscode";
 import configuration from "../configuration";
 
-export class SwiftOutputChannel {
+export class SwiftOutputChannel implements vscode.OutputChannel {
     private channel: vscode.OutputChannel;
+    private logStore = new RollingLog(1024 * 1024 * 5);
+    private logToConsole: boolean;
 
-    constructor() {
-        this.channel = vscode.window.createOutputChannel("Swift");
+    public name: string;
+
+    /**
+     * Creates a vscode.OutputChannel that allows for later retrival of logs.
+     * @param name
+     */
+    constructor(name: string, logToConsole: boolean = true) {
```

Comment:
```suggestion
    constructor(
        public name: string, 
        private logToConsole: boolean = true
    ) {
```
And remove them from above